crates/apollo-mcp-server/src/server/states/running.rs

```diff
@@ -412,34 +493,9 @@ impl ServerHandler for Running {
     async fn list_tools(
         &self,
         _request: Option<PaginatedRequestParam>,
-        _context: RequestContext<RoleServer>,
+        context: RequestContext<RoleServer>,
     ) -> Result<ListToolsResult, McpError> {
-        let meter = &meter::METER;
-        meter
-            .u64_counter(TelemetryMetric::ListToolsCount.as_str())
-            .build()
-            .add(1, &[]);
-
-        Ok(ListToolsResult {
-            next_cursor: None,
-            tools: self
-                .operations
-                .read()
-                .await
-                .iter()
-                .map(|op| op.as_ref().clone())
-                .chain(self.execute_tool.as_ref().iter().map(|e| e.tool.clone()))
-                .chain(self.introspect_tool.as_ref().iter().map(|e| e.tool.clone()))
-                .chain(self.search_tool.as_ref().iter().map(|e| e.tool.clone()))
-                .chain(self.explorer_tool.as_ref().iter().map(|e| e.tool.clone()))
-                .chain(self.validate_tool.as_ref().iter().map(|e| e.tool.clone()))
-                .chain(
-                    self.apps
-                        .iter()
-                        .flat_map(|app| app.tools.iter().map(|tool| tool.tool.clone())),
-                )
-                .collect(),
-        })
+        self.list_tools_impl(context.extensions).await
```

Comment:
> Refactor list_tools to have a list_tools_impl method to make testing easier

What was the specific difficulty with testing list_tools directly? Was it constructing RequestContext<RoleServer> or something else?

I'm a bit hesitant about the `xxx_impl` approach where `list_tools` becomes a one-liner delegating to `list_tools_impl`. The `_impl` suffix doesn't represent a meaningful abstraction and it's purely a testing artifact. We're testing `list_tools_impl`, not `list_tools`. If the delegation breaks, tests will still pass.